src/repr/type.ml

```diff
@@ -366,13 +322,39 @@ let like ?pp ?of_string ?json ?bin ?unboxed_bin ?equal ?compare ?short_hash
         | None -> Structural)
   in
   let pp = to_impl pp
-  and json = to_impl json
+  and json =
+    match json with
+    | Some x -> Custom x
+    | None -> (
+        let rec is_prim : type a. a t -> bool = function
+          | Self s -> is_prim s.self_fix
+          | Attributes { attr_type; _ } -> is_prim attr_type
```

Comment:
What if the attribute is a `Type_pp.Attr_pp`, a `Type_pp.Attr_of_string` or a new fancy attribute that is not yet part of repr?